extension/json/src/json_extension.cpp

```diff
@@ -0,0 +1,49 @@
+#include "json_extension.h"
+
+#include "catalog/catalog.h"
+#include "common/types/types.h"
+#include "json_functions.h"
+#include "json_scan.h"
+#include "main/client_context.h"
+#include "main/database.h"
+
+namespace kuzu {
+namespace json_extension {
+
+#define ADD_FUNC(FUNC_STRUCT)                                                                      \
+    extension::ExtensionUtils::registerFunctionSet(db, std::string(FUNC_STRUCT::name),             \
+        FUNC_STRUCT::getFunctionSet())
+
+void JsonExtension::load(main::ClientContext* context) {
+    auto& db = *context->getDatabase();
+    ADD_FUNC(ToJsonFunction);
+    ADD_FUNC(JsonMergeFunction);
+    ADD_FUNC(JsonExtractFunction);
+    ADD_FUNC(JsonArrayLengthFunction);
+    ADD_FUNC(JsonContainsFunction);
+    ADD_FUNC(JsonKeysFunction);
+    ADD_FUNC(JsonStructureFunction);
+    ADD_FUNC(JsonValidFunction);
+    ADD_FUNC(MinifyJsonFunction);
+    extension::ExtensionUtils::registerTableFunction(db, JsonScan::getFunction());
+    if (!db.getCatalog()->containsType(&transaction::DUMMY_READ_TRANSACTION, "json")) {
```

Comment:
I think createType should be able to handle duplicate json type, so you probably don't need the  check:
```
!db.getCatalog()->containsType(&transaction::DUMMY_READ_TRANSACTION, "json")
```

Reply:
catalog.createType does an assertion instead of handling it
```
    KU_ASSERT(!types->containsEntry(transaction, name));
```